### PR TITLE
Sto 255 admin module error

### PR DESCRIPTION
**Pull request by neildsouth**
Fixes # .

This fixs the issue with the created user credentials

### Status
**Ready**

### Types of changes

- [ x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ x] All tests passed locally by running `./src/run-tests-in-docker.sh`.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog

### CODE REVIEW

File: src/Plugins/MinIO/Monai.Deploy.Storage.MinIO.csproj [hunk 1]
```diff
@@ -52,4 +52,14 @@ SPDX-License-Identifier: Apache License 2.0
   <ItemGroup>
     <Compile Include="..\..\AssemblyInfo.cs" Link="AssemblyInfo.cs" />
   </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="Mc\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="Mc\mc.exe">
```
Comment (mocsharp) [inline src/Plugins/MinIO/Monai.Deploy.Storage.MinIO.csproj:61]: I'm not sure about bundling `mc.exe` given that it's AGPLv3.  I will need verify with legal.

File: src/Plugins/MinIO/StorageAdminService.cs [hunk 2]
```diff
@@ -1,226 +1,262 @@
...
+        {
+            var result = await ExecuteAsync($"admin user remove {_serviceName} {username}").ConfigureAwait(false);
+
+            if (!result.Any(r => r.Contains($"Removed user `{username}` successfully.")))
+            {
+                throw new InvalidOperationException("Unable to remove user");
+            }
+        }
+
+        [Obsolete("CreateUserAsync with bucketNames is deprecated, please use CreateUserAsync with an array of PolicyRequest instead.")]
```
Comment (mocsharp) [inline src/Plugins/MinIO/StorageAdminService.cs:176]: Should we just get rid of the method since it's only used by Task Manager?

Comment (neildsouth) [reply src/Plugins/MinIO/StorageAdminService.cs:176]: Thought we'd keep it just for a while for people still referencing older nuget package ?

**Review by mocsharp — APPROVED** [review]

File: src/Storage/API/IStorageAdminService.cs [hunk 3]
```diff
@@ -27,6 +28,15 @@ public interface IStorageAdminService
...
+        [Obsolete("CreateUserAsync with bucketNames is deprecated, please use CreateUserAsync with an array of PolicyRequest instead.")]
         Task<Credentials> CreateUserAsync(string username, AccessPermissions permissions, string[] bucketNames);
+
+        /// <summary>
+        /// Creates a user with read only permissions
+        /// </summary>
+        /// <param name="username">Username</param>
+        /// <param name="policyRequests">Contains the buckets and folders that the user needs access to</param>
+        /// <returns></returns>
+        Task<Credentials> CreateUserAsync(string username, PolicyRequest[] policyRequests);
```
Comment (mocsharp) [inline src/Storage/API/IStorageAdminService.cs:40]: May I ask another favor? Can we make this `params PolicyRequest[] policyRequests` instead? 😁

Comment (neildsouth) [reply src/Storage/API/IStorageAdminService.cs:40]: sure :)

**Review by jackschofield23 — APPROVED** [review]

**Pull request merged**